### PR TITLE
update

### PR DESCRIPTION
**Pull request by charneff**
Duplicative language around has many/belongs ro removed. This was confusing students. Added in restful routing and proper state updating. Also added in validations and error messaging.

### CODE REVIEW

**Review by lizbur10 — COMMENTED** [review]
I think we need to hold off on merging this one in for a week or two. I'll need to get `useContext` out of the optional module before this goes live. There's also the Heroku issue - they have announced a student option and we're looking into whether that's going to work for us. In the meantime, it's fine to make the edit on lines 207-208 for current cohorts.

**Comment by charneff** [comment]
Gotcha,. That is totally fine. Whenever updated to required, would love to add in useContext =)

File: README.md [hunk 1]
```diff
@@ -24,22 +24,33 @@ For this project, you must:
...
     submittable attribute.
   - Full CRUD actions for at least one resource.
   - Minimum of create and read actions for EACH resource.
+- Follow RESTful routing convention for backend routes.
+- [Active Record validations](https://guides.rubyonrails.org/active_record_validations.html) must be present on your models for most
+  attributes.
+- Use controller validations to alter back end json response to front end. 
+  The response should pass your object if the creation, update, or deletion
+  succeeds. However, the response should pass error messages to the front 
+  end and displayed if the action fails. HINT: Utilize `record.errors`.
```
Comment (lizbur10) [inline README.md]: Correction: "...and display them if ..."

File: README.md [hunk 2]
```diff
@@ -193,9 +204,8 @@ features instead of a solid core of working features.
 ## Deploying
 
 The template project has all the starter code needed to help you deploy your
-application to Heroku. It's recommended to deploy your project early and push up
-changes often to ensure that your code works equally well in production and
-development environments.
+application to Heroku. Please note that Heroku is sunsetting their free tier 
+hosting abilities as of November. We do not require deployment of your project.
```
Comment (lizbur10) [inline README.md]: This change needs to be undone. All that needs to change is Heroku -> Render

**Review by lizbur10 — CHANGES_REQUESTED** [review]
Hi @charneff! So sorry this slipped through the cracks!!

I've requested a couple of corrections then I think we can merge it in

**Review by lizbur10 — APPROVED** [review]

**Pull request merged**